### PR TITLE
Fix intensity calculations if axis order is swapped

### LINKED ISSUE #283: Intensity options breaks when axes are swapped
Trying to change the intensity option when energy transfer is on the x axis results in an error:
```
operands could not be broadcast together with shapes (294,1) (604,294)
```

### PR DESCRIPTION
**Pull request by martyngigg**
Description of work.

See commit message.

**To test:**

* Load a file and plot a slice with the energy axis as the Y axis and check the intensity operations.
* Now repeat  the operations with the axes swapped on the slice. 

Fixes #283.

### CODE REVIEW

File: mslice/models/slice/mantid_slice_algorithm.py [hunk 1]
```diff
@@ -90,21 +90,44 @@ def _compute_slice_nonPSD(self, workspace, x_axis, y_axis, norm_to_one):
...
-    def compute_boltzmann_dist(self, sample_temp, e_axis):
+    def axis_values(self, axis, reverse=False):
+        """
+        Compute a set of bins for the given axis values
+        :param axis: Axis object defining axis details
+        :param reverse: If true then the axis should have values in decreasing order
+        :return: A new numpy array containing the axis values
+        """
+        if reverse:
+            values = np.linspace(axis.end, axis.end, self._get_number_of_steps(axis))
```
Comment (mducle) [inline mslice/models/slice/mantid_slice_algorithm.py]: Should this be `np.linspace(axis.end, axis.start, self._get_number_of_steps(axis))`

**Comment by MatthewBowles** [comment]
Strangely it seems like changing intensity to Chi'' magnetic and then to d2sigma while energy is on the x axis causes an error.

File: mslice/models/slice/mantid_slice_algorithm.py (hunk 1 above)
Comment (martyngigg) [reply mslice/models/slice/mantid_slice_algorithm.py]: Yes it should :man_facepalming:. I was trying to be fancy with numpy at one point and put it back wrong. Thanks for spotting that!

**Comment by MatthewBowles** [comment]
GDOS and Symmetrised are still broken for me

File: mslice/models/slice/mantid_slice_algorithm.py [hunk 2]
```diff
@@ -179,8 +179,10 @@ def compute_gdos(self, scattering_data, sample_temp, q_axis, e_axis, data_rotate
         momentum_transfer = np.square(momentum_transfer, out=momentum_transfer)
         boltzmann_dist = self.compute_boltzmann_dist(sample_temp, energy_transfer)
         if data_rotated:
-            gdos = scattering_data / momentum_transfer[None, :]
-            gdos *= energy_transfer[None, :]
+            scattering_rotated = np.swapaxes(scattering_data, 0, 1)
+            gdos = scattering_rotated / momentum_transfer
+            gdos = np.swapaxes(gdos, 0, 1)
```
Comment (martyngigg) [inline mslice/models/slice/mantid_slice_algorithm.py]: Nice :+1:

**Comment by MatthewBowles** [comment]
I don't think these are optimal fixes but everything **should** work without errors now.

File: mslice/models/slice/mantid_slice_algorithm.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def compute_symmetrised(self, scattering_data, sample_temp, e_axis, data_rotated
         else:
             rhs = scattering_data[-negative_de_len:, :] * boltzmann_dist[:, None]
             lhs = scattering_data[:-negative_de_len, :]
-            return np.concatenate((rhs, lhs))
+            return np.concatenate((lhs, rhs))
```
Comment (martyngigg) [inline mslice/models/slice/mantid_slice_algorithm.py]: How did I manage to get those the wrong way around :man_facepalming: :face_with_head_bandage:

**Comment by martyngigg** [comment]
@MatthewBowles Thanks! Please create an issue if you think it's worth improving.

**Pull request merged**